samples/arduino/starterkit/Spaceship2.js

```diff
@@ -41,31 +43,34 @@ var btnstate = false;
 // turn green LED on initially
 led1.write(true);
 
-var timer;
+var timer = null;
 var toggle;
+var last;
 
 button.onchange = function (event) {
     if (event.value) {
-        // button has been pressed
-        toggle = false;
-        led1.write(false);
-        led2.write(false);
-        led3.write(true);
+        if (timer == null) {
```

Comment:
I had to put this in because, even using JUST Zephyr samples, a GPIO input can trigger the same value multiple times. If this happens, this top if block will get called causing two `setInterval`'s to trigger. For this reason I set `timer` to null once the interval is cleared. This provides synchronization between `setInterval` and `clearInterval`